### PR TITLE
BDRSPS-557 Create survey metadata instructions

### PR DESCRIPTION
**Pull request by joecrowleygaia**
Created survey metadata instruction template and rendering logic. Made rendering contexts registerable and retrievable via the cli invocation for generating instructions.
Fixes also a few minor formatting issues noticed in other template instructions.

### CODE REVIEW

File: abis_mapping/templates/incidental_occurrence_data_v2/templates/instructions.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ datatype format, and examples.</ins>
 
 ## APPENDICES
 ### APPENDIX-I: VOCABULARY LIST
-Apart from geodeticDatum, the data validation does not require adherence tho the below vocabularies
+Apart from geodeticDatum, the data validation does not require adherence to the below vocabularies
 for each of the fields indicated as having vocabularies. These vocabularies are provided as a
```
Comment (chungvl) [inline abis_mapping/templates/incidental_occurrence_data_v2/templates/instructions.md:72]: "Apart from geodeticDatum, the fields indicated as having vocabularies are not required to adhere to their corresponding vocabulary." ?

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# SYSTEMATIC SURVEY METADATA TEMPLATE INSTRUCTIONS
+
+## OVERVIEW
+Use this template to record systematic survey metadata; that is metadata relating to a
+Systematic Survey dataset.
+
+This Systematic Survey Metadata template must be used in combination with the Systematic
+Survey Occurrence template and, in some cases, the Systematic Survey Site template.
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:8]: "Survey Occurrence template and, in some cases, the Systematic Survey Site template." -> "Survey Occurrence template and in some cases, the Systematic Survey Site template." ?

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+Templates have been provided to facilitate integration of your data into the Biodiversity
+Data Repository database. Not all types of data have been catered for in the available
+templates at this stage; therefore, if you are unable to find a suitable template, please
+contact <bdr-support@gaiaresources.com.au> to make us aware of your data needs.
+
+#### NEED TO KNOW:
+For data validation, you will need your data file to:
+- be the correct **file format**,
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:17]: be in the correct...

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+- have **matching template fields** to the template downloaded (do not remove, or
+change the order of fields),
+- additional fields may be added **after the templated fields** (noting that the data type
+is not assumed and values will be encoded as strings),
+- only one row of metadata should be included and only the first row of metadata
+**will be accepted** (this symbolises one Survey per dataset submission).
+- have values in **mandatory fields** (see Table 1), and
+- comply with data **value constraints,** for example the geographic coordinates are
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:25]: brackets around the example?

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+consistent with a geodeticDatum type of the {{values.geodetic_datum_count}} available options.
+
+### FILE FORMAT
+- The systematic survey metadata template is a [UTF-8](#appendix-iii-utf-8) encoded csv (not Microsoft
+Excel Spreadsheets). Be sure to save this file with your data as a .csv (UTF-8) as
+follows, otherwise it will not pass the in-browser csv validation step upon upload.
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:31]: CSV

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+- **Do not include empty rows.**
+
+#### FILE SIZE
+MS Excel imposes a limit of 1,048,576 rows on a spreadsheet, limiting a CSV file to the
+header row followed by 1,048,575 occurrences. Furthermore, MS Excel has a 32,767
+character limit on individual cells in a spreadsheet. These limits may be overcome by using
+or editing CSV files with other software.
+
+## TEMPLATE FIELDS
+The template file contains the field names in the top row that form part of the core Survey
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:43]: first row?

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+data model. Table 1 will assist you in transferring your data to the template with the following
+information:
+- **Field name** in the template (and an external link to the Darwin Core standard for that
+field where available);
+- **Description** of the field;
+- **Required** i.e. whether the field is **mandatory or optional;**
+- **Datatype format** required for the data values for example text (string), number
+(integer, float), or date; and
+- **Example/s** of an entry for that field.
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:52]: Example(s)

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+- **Vocabulary links** within this document (for example pick list values) where relevant.
+The fields that have suggested values options for the fields in Table 1 are listed in
+Table 2 in alphabetical order of field name.
+
+### ADDITIONAL FIELDS
+Data that do not match the existing template fields may be added as additional columns in
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:58]: Data that does not match

File: abis_mapping/templates/survey_metadata/templates/instructions.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+E.g., sampleSizeUnit, sampleSizeValue.
+
+<ins>Table 1: Systematic Survey Metadata template fields with descriptions, conditions,
+datatype format, and examples.</ins>
+
+{{tables.fields}}
+
+## APPENDICES
+### APPENDIX-I: VOCABULARY LIST
+Apart from geodeticDatum, the data validation does not require adherence to the below vocabularies
```
Comment (chungvl) [inline abis_mapping/templates/survey_metadata/templates/instructions.md:69]: See earlier suggestion

**Review by chungvl — CHANGES_REQUESTED** [review]
Did some proofreading of the text. See comments.

File: abis_mapping/templates/survey_metadata/templates/instructions.md (hunk 3 above)
Comment (joecrowleygaia) [reply abis_mapping/templates/survey_metadata/templates/instructions.md:17]: matches existing pdf wording.

**Comment by joecrowleygaia** [comment]
reviewed all suggestions and all within the original pdf instructions, developed by others.

**Review by chungvl — APPROVED** [review]

**Pull request merged**